cpp/src/link_analysis/pagerank_1D.cu

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Author: Alex Fender afender@nvidia.com
+
+#include <graph.hpp>
+#include "pagerank_1D.cuh"
+#include "utilities/graph_utils.cuh"
+
+namespace cugraph {
+namespace opg {
+
+#define CUDA_MAX_KERNEL_THREADS 256
+#define CUDA_MAX_BLOCKS 65535
+
+template <typename VT, typename ET, typename WT>
+__global__ void __launch_bounds__(CUDA_MAX_KERNEL_THREADS)
+  transition_kernel(const size_t e, const VT *ind, const VT *degree, WT *val)
+{
+  for (auto i = threadIdx.x + blockIdx.x * blockDim.x; i < e; i += gridDim.x * blockDim.x)
+    val[i] = 1.0 / degree[ind[i]];
+}
+
+template <typename VT, typename ET, typename WT>
+Pagerank<VT, ET, WT>::Pagerank(const raft::handle_t &handle_,
+                               experimental::GraphCSCView<VT, ET, WT> const &G)
+  : comm(handle_.get_comms())
```

Comment:
Eventually we need to adopt RAFT's clang-tidy.

https://github.com/rapidsai/raft/blob/3038c773636540941b8bf9e30c487f20bddc3cdc/cpp/.clang-tidy#L185

Class private member variables should have a suffix "_"

Better adopt this convention in new code.

Reply:
Do you know of a tool that does this on the code base like for clang?

Reply:
clang-tidy does this. This isn't set-up for cuGraph but for RAFT, python3 cpp/scripts/run-clang-tidy.py -cdb cpp/biuld/compile_commands.json does this.

Reply:
Let's proceed as we did for clang support then and port everything at once. 
Once this is done and checks and tools are enabled, new PRs will be requested to comply with tidy.

Reply:
https://github.com/rapidsai/cugraph/issues/958